buildingmotif/ingresses/base.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from functools import cached_property
+from typing import List
+
+from rdflib import Graph, Namespace
+
+from buildingmotif import BuildingMOTIF
+
+
+@dataclass
+class Record:
+    """Represents a piece of metadata from some metadata ingress"""
+
+    rtype: str
+    fields: dict
+
+
+class RecordIngressHandler:
+    """Generates Record instances from an underlying metadata source"""
+
+    def __init__(self, bm: BuildingMOTIF):
+        self.bm = bm
+
+    @cached_property
+    def records(self) -> List[Record]:
+        """
+        Generates (then caches) a list of Records from an underlying data source
+        """
+        raise NotImplementedError("Must be overridden by subclass")
+
+
+class GraphIngressHandler:
```

Comment:
I think we can still keep an abstract super class `IngressHandler` for both of these, but this looks good.

Reply:
Added!